### PR TITLE
python3Packages.authcaptureproxy: init at 0.4.2

### PR DESCRIPTION
**Pull request by graham33**
###### Motivation for this change

Required for new version of teslajsonpy (PR coming soon).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/authcaptureproxy/default.nix [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  disabled = !isPy3k;
+
+  # Lower importlib-metadata requirement and remove entirely Python >= 3.8 since
+  # it's built-in (and the code will use the built-in version)
+  #
+  # TODO: update once we have importlib-metadata >= 3.4.0 in nixpkgs
+  patchPhase = if pythonAtLeast "3.8" then ''
+    sed -i '/importlib-metadata/d' pyproject.toml
+  '' else ''
+    sed -i 's/importlib-metadata = "^3.4.0"/importlib-metadata = "^1.7.0"/' pyproject.toml
```
Comment (graham33) [inline pkgs/development/python-modules/authcaptureproxy/default.nix:40]: Upgrading importlib-metadata is quite an invasive change, so I decided not to take that on here.

Comment (rmcgibbo) [reply pkgs/development/python-modules/authcaptureproxy/default.nix:40]: Looks like that's #113028

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113855` at 47a260ce run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/8177867633dfda19ebb2b89b36116d28">python37Packages.authcaptureproxy</a></li></li>
    <li><a href="https://gist.github.com/af3d386906b662834ff0f9bb2e367773">python38Packages.authcaptureproxy</a></li></li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113855` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python37Packages.authcaptureproxy</li>
<li>python38Packages.authcaptureproxy</li>
</ul>
</details>

**Comment by graham33** [comment]
I think this should be good to go, if anyone has time to review.  I can update it when #113028 is resolved.

**Comment by mweinelt** [comment]
Superseded by #121457 with a more recent version of authcaptureproxy.

**Pull request closed without merging**